src/OpenTelemetry/Metrics/HistogramBuckets.cs

```diff
@@ -36,9 +40,38 @@ public class HistogramBuckets
 
         internal int IsCriticalSectionOccupied = 0;
 
+        private readonly Bucket root;
+
+        private readonly Func<double, int> findHistogramBucketIndex;
+
         internal HistogramBuckets(double[] explicitBounds)
         {
             this.ExplicitBounds = explicitBounds;
+            this.findHistogramBucketIndex = this.FindBucketIndexLinear;
```

Comment:
This would probably cause allocation and create a closure for using `this.ExplicitBounds`.

Reply:
Added to histogram improvement issue https://github.com/open-telemetry/opentelemetry-dotnet/issues/3428